File: src/main/kotlin/com/itmo/microservices/demo/order/impl/service/IOrderService.java
package com.itmo.microservices.demo.order.impl.service;

import com.itmo.microservices.demo.order.api.dto.Booking;
import com.itmo.microservices.demo.order.api.dto.Order;

import java.io.IOException;
import java.net.MalformedURLException;
import java.net.ProtocolException;
import java.util.UUID;

public interface IOrderService {
    Order createOrder(Order order);
    Order getOrderById(UUID orderId);
    void putItemToOrder(UUID orderId, UUID itemId, int amount);
    Booking book(UUID orderId) throws IOException;
    void selectDeliveryTime(UUID orderId, int seconds) throws IOException;
}

File: src/main/kotlin/com/itmo/microservices/demo/order/impl/service/IOrderService.kt
package com.itmo.microservices.demo.order.impl.service

import com.itmo.microservices.demo.order.api.dto.Booking
import com.itmo.microservices.demo.order.api.dto.Order
import java.util.*

interface IOrderService {
    fun createOrder(order: Order?): Order?
    fun getOrderById(orderId: UUID?): Order?
    fun updateOrder(orderId: UUID?, itemId: UUID?, amount: Int)
    fun book(orderId: UUID?): Booking?
    fun selectDeliveryTime(orderId: UUID?, seconds: Int)
}